### PR TITLE
A small projection on panels, and use sets instead of explicit loops

### PR DESCRIPTION
**Pull request by dnil**
This PR speeds up a slow controller, partially addressing #4655.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by DN, also automated tests since this is supposed to be just a refactor

### CODE REVIEW

File: scout/server/blueprints/cases/controllers.py [hunk 1]
```diff
@@ -557,19 +563,17 @@ def check_outdated_gene_panel(panel_obj, latest_panel):
         missing_genes, extra_genes
     """
     # Create a list of minified gene object for the case panel {hgnc_id, gene_symbol}
-    case_panel_genes = [
```
Comment (dnil) [inline scout/server/blueprints/cases/controllers.py:560]: Is there any real reason for the full comparison of these gene dicts? I have a feeling they used to be passed back to the controller at some point..

**Review by northwestwitch — APPROVED** [review]
Looks good. Nice with sets. Have you tested it?

**Comment by dnil** [comment]
Ive tested, in as much as it is just supposed to pass tests and do what the previous things did, but not really timed. I can merge it into the timing branch perhaps and see what we get?

**Comment by dnil** [comment]
I think it shaves a decent slice; looks like the panels are stably below matching causatives now on the case controller timing branch with this merged in:
Screenshot 2024-06-04 at 10 40 17

**Pull request merged**